timelock-impl/src/main/java/com/palantir/atlasdb/timelock/adjudicate/LeaderElectionMetricAggregator.java

```diff
@@ -16,39 +16,41 @@
 
 package com.palantir.atlasdb.timelock.adjudicate;
 
-import com.palantir.atlasdb.util.CurrentValueMetric;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.SlidingTimeWindowArrayReservoir;
 import com.palantir.atlasdb.util.MetricsManager;
 import com.palantir.atlasdb.util.SlidingWindowWeightedMeanGauge;
-import com.palantir.conjure.java.lib.SafeLong;
 import com.palantir.timelock.feedback.LeaderElectionStatistics;
+import java.util.concurrent.TimeUnit;
 
 public final class LeaderElectionMetricAggregator {
     private final SlidingWindowWeightedMeanGauge weightedGaugeP99;
     private final SlidingWindowWeightedMeanGauge weightedGaugeP95;
     private final SlidingWindowWeightedMeanGauge weightedGaugeMean;
-    private final CurrentValueMetric<Long> leaderElectionEstimate;
+    private final LeaderElectionDurationAccumulator leaderElectionDurationAccumulator;
 
     public LeaderElectionMetricAggregator(MetricsManager metricsManager) {
         weightedGaugeP99 = SlidingWindowWeightedMeanGauge.create();
         weightedGaugeP95 = SlidingWindowWeightedMeanGauge.create();
         weightedGaugeMean = SlidingWindowWeightedMeanGauge.create();
-        leaderElectionEstimate = new CurrentValueMetric<>();
         metricsManager.registerMetric(
                 LeaderElectionMetricAggregator.class, "leaderElectionImpactMean", weightedGaugeMean);
         metricsManager.registerMetric(
                 LeaderElectionMetricAggregator.class, "leaderElectionImpactP95", weightedGaugeP95);
         metricsManager.registerMetric(
                 LeaderElectionMetricAggregator.class, "leaderElectionImpactP99", weightedGaugeP99);
-        metricsManager.registerMetric(
-                LeaderElectionMetricAggregator.class, "leaderElectionEstimate", leaderElectionEstimate);
+        Histogram leaderElectionHistogram = metricsManager.registerOrGetHistogram(
+                LeaderElectionMetricAggregator.class,
+                "leaderElectionDurationEstimate",
+                () -> new Histogram(new SlidingTimeWindowArrayReservoir(5, TimeUnit.MINUTES)));
+        leaderElectionDurationAccumulator = new LeaderElectionDurationAccumulator(leaderElectionHistogram::update, 10);
```

Comment:
nit: can we extract a constant for this number? Possibly above too although for some reason I don't mind as much for the reservoir.